spack-environments/archer2/compute-node/spack.yaml

```diff
@@ -132,10 +132,10 @@ spack:
   - compiler:
       spec: gcc@9.3.0
       paths:
-        cc: cc
-        cxx: CC
-        f77: ftn
-        fc: ftn
+        cc: /opt/cray/pe/gcc/9.3.0/snos/bin/gcc
+        cxx: /opt/cray/pe/gcc/9.3.0/snos/bin/g++
+        f77: /opt/cray/pe/gcc/9.3.0/snos/bin/gfortran
+        fc: /opt/cray/pe/gcc/9.3.0/snos/bin/gfortran
```

Comment:
@giordano If you use the bare compiler commands in this way (rather than the wrappers) then there is no guarantee things will work as expected. The HPE Cray Programming Environment is all built around the design choice that you never call the compilers directly but, instead, always access them through the wrapper commands. Your approach may work if you set all the library spec (as you have done below) but you may encounter weirdness as they were never designed to be used like this.

Reply:
The main motivation for going directly to the bare GCC binaries is that yesterday I struggled to build a package using the Cray wrappers for GCC, more specifically Spack wasn't able to detect the compilers implicit rpaths
```console
> spack build-env 'grid@develop%gcc' -- printenv SPACK_COMPILER_IMPLICIT_RPATHS
>
```
because the Cray wrappers were trying, and failing, to lo load some extra modules.  Instead I had no problems with the environment in this PR:
```console
> spack build-env 'grid@develop%gcc' -- printenv SPACK_COMPILER_IMPLICIT_RPATHS
/opt/cray/pe/gcc/11.2.0/snos/lib/gcc/x86_64-suse-linux/11.2.0:/opt/cray/pe/gcc/11.2.0/snos/lib64
>
```
Getting the implicit RPATHs is important because they contain the directory where libstdc++, and trying to run a C++ program built with GCC 11 (like the one I was trying to build yesterday) without the RPATH set correctly would fail because it'd try to use the global libstdc++, which has an older ABI.

If you're specifically worried about Cray MPICH, note that since version 8.1.7 this package comes with its own compiler wrappers, and Spack automatically uses them when available: https://github.com/spack/spack/blob/b21e54dffa7b666462f063ab9d4d51109202b956/var/spack/repos/builtin/packages/cray-mpich/package.py#L32-L33.  This is also the reason why I added to the environment `cray-mpich` v8.1.5.  I verified the right libmpi is being linked in this case, and the application runs fine when launched with `srun`.